google_sso/README.md

```diff
@@ -0,0 +1,307 @@
+
+Select an E2-medium instance with the Ubuntu 18.04 LTS image and a 100GB HDD.
+
+_Note that the machine specs are mostly artbitrary - HAS doesn't require powerful VMs
+but you might consider more powerful specs if you were to actually run Helix Core in production._
+
+![Create Instance](create_instance.png)
+
+Alternatively, you can create the instance using the gcloud command:
+
+```
+gcloud compute instances create perforce --zone=us-central1-a --machine-type=e2-medium --image=ubuntu-1804-bionic-v20210129 --image-project=ubuntu-os-cloud --boot-disk-size=100GB --boot-disk-type=pd-standard
+```
+
+Please note that for the purposes of keeping this simple, the VM has been configuring using
+an ephemeral external IP. In an actual production environment, you'll probably want to assign
+a permanent external IP and use it in conjuction with a custom domain name.
+
+Take a note of the VM's external IP address, you will need it in the subsequent steps.
+We will refer to is as VM_EXTERNAL_IP in the rest of the guide.
+
+### Configuring OAuth
+
+In order to use HAS with Google Identity, you will need to configure an OAuth 2.0 client.
+In the Google Cloud Console, navigate to API & Services and select Credentials.
+Click on Create Credentials, select "OAuth client ID" and pick "Web application"
+as the Application type:
+
+![Create Credentials](create_credentials.png)
+
+You can name the credentials "HAS Client" (the name does not matter).
+
+Note that you will need to provide the authorized Authorized redirect URIs.
+Since we are not using a domain name but an IP address, we would need to use a special domain,
+xip.io, that does the redirect. You can see the following post for more details:
+https://stackoverflow.com/questions/14238665/can-a-public-ip-address-be-used-as-google-oauth-redirect-uri
+
+The two URIs you'll need to list are:
+
+-  https://VM_EXTERNAL_IP.xip.io:3000
+-  https://VM_EXTERNAL_IP.xip.io:3000/oidc/callback
+
+If this the first time you are creating OAuth credentials, you will also need to configure
+an OAuth consent screen.
+
+Once you have created the credentials, take note of the Client ID and the Client Secret.
+These values will be needed when configuring HAS.
+
+### Configuring the firewall
+
+We will most likely want to access the Perforce VM from outside of the default VPC network.
+For the purpose of this exercise, we will open up the VPC firewall to allow unrestricted
+access to our VMs on ports 1666 (Helix Core) and 3000 (HAS).
+
+**Please consult a security or network engineer when adapting this to your environment.**
+
+To open up all VMs on your GCP network to incoming traffic on ports 1666 and 3000, follow these steps:
+
+-  In Google Cloud Console, navigate to VPC Network / Firewall
+-  Click on "Create Firewall Rule"
+
+![Create Firewall](create_firewall.png)
+
+You can give your rule any name (e.g. "helix-and-has"). Select "All instances in the network"
```

Comment:
Two separate firewall rules, one for Helix (1666/tcp) and another for HAS (3000/tcp) are more appropriate use of least privilege principle.

Reply:
Done